### ISSUE TITLE
HTML inside of strings in if statement cause false positive for types.

### ISSUE THREAD

**Issue by envygeeks** [Microsoft/TypeScript#13065]
**TypeScript Version:**  2.1.4
**Code**

```ts
if (converted !== "<p></p>" || converted !== "<p><br></p>") {
  // Broken
}
```

**Expected behavior:** Properly outputted Typescript.
**Actual behavior:**

```
ERROR in ./path.ts
(25,36): error TS2365: Operator '!==' cannot be applied to types '"<p></p>"' and '
  "<p><br></p>"'
```

**Comment by DanielRosenwasser** [comment]
If `converted` is not equal to `"<p></p>"`, then you'll jump into the body of the `if`. If it is equal to that, then you're going to check whether it's equal to `"<p><br></p>"`, which it clearly can't be, because TypeScript just determined that it's equal to `"<p></p>"`.

So did you mean to write `&&`?

**Comment by envygeeks** [comment]
That's nonsense.  Sure there was a mistake within the example (and not within the source) but you're telling me that instead of properly syntax checking and spitting out a valid error it automatically jumps to the wrong conclusion that a string must be a type and that's valid?  Regardless of the mistake that was made it shouldn't have even come close to that conclusion based on the context around it, how does that even make sense?

**Comment by RyanCavanaugh** [comment]
What exactly is the bug report here?

**Comment by envygeeks** [comment]
There is no bug here, I'll move along and deal with it.
